### PR TITLE
add prog_avail_local function

### PR DESCRIPTION
**Pull request by ykakarap**
This PR adds `prog_avail_local` support for crashd scripts.
`prog_avail_local` check if a program is available on host machine.

Example script:
```
def run():
    prog="echo"
    path = prog_avail_local(prog=prog)
    if path == "":
        print("prog '{}' is not available".format(prog))
    else:
        print("prog '{}' is available at '{}'".format(prog, path))

run()
```

### CODE REVIEW

File: starlark/prog_avail_local_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+				if test.exists {
+					t.Fatalf("expecting '%v' to exists but 'prog_avail_local' didnt find it", test.prog)
+				} else {
+					t.Fatalf("expecting '%v' to not exists but 'prog_avail_local' found it", test.prog)
+
+				}
+			}
+		})
+	}
+}
```
Comment (vladimirvivien) [inline starlark/prog_avail_local_test.go:60]: Add tests to run the code from starlark script as is done here for instance

https://github.com/vmware-tanzu/crash-diagnostics/blob/8b79e7dc6ecf1dacbfa1270eb7f7946409269a8c/starlark/run_local_test.go#L45

**Review by vladimirvivien — CHANGES_REQUESTED** [review]
This looks good, but needs additional test to exercise the actual script

File: starlark/prog_avail_local_test.go (hunk 1 above)
Comment (ykakarap) [reply starlark/prog_avail_local_test.go:60]: The test is running and testing the using a starlark script. In [this line](https://github.com/ykakarap/crash-diagnostics/blob/prog_avail/starlark/prog_avail_local_test.go#L34) you can see that it is creating a simple script to test the `prog_avail_local` function.

Comment (vladimirvivien) [reply starlark/prog_avail_local_test.go:60]: @ykakarap oh I see now.  You should consider adding the script in the test struct.  That way you can test different scenario such as 
```go
		{
			name:   "prog_avail_local checks for 'go'",
			prog:   "go",
			exists: true,
                      script: "prog_avail_local(prog="go")
		},
		{
			name:   "prog_avail_local default arg",
			prog:   "go",
			exists: true,
                      script: "prog_avail_local("go")
		},
		{
			name:   "prog_avail_local full path",
			prog:   "go",
			exists: true,
                      script: "prog_avail_local("/full/path/to/go")
		},
```

Comment (ykakarap) [reply starlark/prog_avail_local_test.go:70]: @vladimirvivien I have updated tests

**Review by vladimirvivien — APPROVED** [review]
LGTM!!!

**Pull request merged**